torch/distributed/_tensor/device_mesh.py

```diff
@@ -164,18 +166,19 @@ def _init_process_groups(self):
                 f"DeviceMesh cannot have duplicate values, but found {self.mesh.tolist()}"
             )
         # validate that all calling ranks pass in the same `mesh` argument.
-        self_mesh = self.mesh.to(self.device_type)
-        mesh_tensor = funcol.all_gather_tensor(
-            self_mesh, gather_dim=0, group=_get_default_group()
-        )
-        mesh_tensor_chunked = torch.chunk(mesh_tensor, get_world_size())
-        for other_rank, other_mesh in enumerate(mesh_tensor_chunked):
-            if not torch.equal(self_mesh, other_mesh):
-                raise RuntimeError(
-                    f"DeviceMesh initialization does not allow different mesh argument:"
-                    f"rank {get_rank()} has mesh {self_mesh} while rank {other_rank}"
-                    f"has mesh {other_mesh}!"
-                )
+        if self._validate_mesh:
+            self_mesh = self.mesh.to(self.device_type)
```

Comment:
Maybe we can group this allgather check + the `unique_mesh_values` check together into a private function and calls it under the `validate_mesh` flag

Reply:
Thanks for reviewing, @wanchaol. Make a few updates to this PR addressing the comments:
1. Directly pass along _validate_mesh to init_process_groups().
2. Group allgather check with the unique_mesh_values check in validate_mesh().

Could you review the PR again? 
Will modify skipping pg creation if existed for both 1D and 2D cases and then delete _init_process_groups flag in a follow up PR.